src/server/active_tran_server.cpp

```diff
@@ -276,8 +276,17 @@ void
 active_tran_server::disconnect_page_server ()
 {
   assert_is_active_tran_server ();
+  std::string msg = "" + static_cast<int> (cubcomm::server_server::CONNECT_ACTIVE_TRAN_TO_PAGE_SERVER);
+  er_log_debug (ARG_FILE_LINE, "Transaction server starts disconnecting from the page servers.");
 
+  for (int i = 0; i < m_page_server_conn_vec.size(); i++)
+    {
+      er_log_debug (ARG_FILE_LINE, "Transaction server disconnected from page server with channel id: %s.\n",
+		    m_page_server_conn_vec[i]->get_underlying_channel_id ());
+      push_request (i, ats_to_ps_request::SEND_DISCONNECT_MSG, std::move (msg));
```

Comment:
- Do not change push_request() function. Access m_page_server_conn_vec directly.
- If there are multiple page servers, moving the message on the first request can cause invalid memory access in the other requests. Copy it instead.
```suggestion
      m_page_server_conn_vec[i]->push (ats_to_ps_request::SEND_DISCONNECT_MSG, msg);
```

Reply:
Done.